### PR TITLE
Add random zip file example question

### PR DESCRIPTION
**Pull request by eliotwrobson**
Added an example question to resolve https://github.com/PrairieLearn/PrairieLearn/issues/336 and resolve https://github.com/PrairieLearn/PrairieLearn/issues/1758. Still have to fix an issue where the generated zip file isn't being given to the student (the downloaded file right now is blank for some reason), but the generation logic itself works.

@jonatanschroeder do you have any idea why this might be happening? As I recall, you've used the file generation before.

EDIT: For reasons I cannot fathom, this was happening because the options given to `file()` did not include paths on the server. This was very hard to debug because console output from the `file` function is suppressed.

### CODE REVIEW

**Comment by jonatanschroeder** [comment]
> EDIT: For reasons I cannot fathom, this was happening because the options given to `file()` did not include paths on the server.

Note that the question directory is the current working directory (unless your code changes it for some reason), so you can use a relative path instead. No need to use the paths info in your example. But I agree that paths should probably be provided in file() to allow it to use info from, e.g., clientFilesCourse.

> This was very hard to debug because console output from the `file` function is suppressed.

About the console output: it is not suppressed, it is added to an issue. That said, issues created by file() are typically generated after the page has already been rendered (since it's called from a separate request from the browser), so they don't show up in the page itself right away. You need to either reload the page with the same variant (e.g., by saving a submission) to see the issue, or go to the issues page.

**Comment by eliotwrobson** [comment]
> Note that the question directory is the current working directory (unless your code changes it for some reason), so you can use a relative path instead. No need to use the paths info in your example. But I agree that paths should probably be provided in file() to allow it to use info from, e.g., clientFilesCourse.

Ahh, I did not know this. This is especially confusing because there aren't many guarantees on where files are supposed to be located. So generally, all code I write for PL uses the path info taken from options. Other people have reported this issue before: https://github.com/PrairieLearn/PrairieLearn/issues/5372

> About the console output: it is not suppressed, it is added to an issue. That said...

Ahh, I missed this. I'll double-check the issue list.

File: exampleCourse/courseInstances/SectionA/assessments/homework/extraCredit/infoAssessment.json [hunk 1]
```diff
@@ -38,7 +38,8 @@
                 {"id": "demo/calculation",          "points": 1, "maxPoints": 3},
                 {"id": "demo/randomCheckbox",       "points": 1, "maxPoints": 3},
                 {"id": "demo/randomPlot",           "points": 1, "maxPoints": 3},
-                {"id": "demo/randomFakeData",       "points": 1, "maxPoints": 3}
+                {"id": "demo/randomFakeData",       "points": 1, "maxPoints": 3},
+                {"id": "demo/randomZippedFile",       "points": 1, "maxPoints": 3}
```
Comment (jonatanschroeder) [inline exampleCourse/courseInstances/SectionA/assessments/homework/extraCredit/infoAssessment.json:42]: Nit: 
```suggestion
                {"id": "demo/randomFakeData",       "points": 1, "maxPoints": 3},
                {"id": "demo/randomZippedFile",     "points": 1, "maxPoints": 3}
```

File: exampleCourse/questions/demo/randomZippedFile/question.html [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  </p>
+  <p>
+    Specifically, we generate a zip file <code>generated_files.zip</code>, containing two files,
+    <code>{{params.text_file_name}}</code>, whose contents are
+    <code>{{params.words_for_text}}</code>, and <code>{{params.python_file_name}}</code>, which
+    contains a function named <code>{{params.function_name}}</code> that prints
+    <code>Hello world!</code>.
+  </p>
+  <p>
+    Download the zipped file here
```
Comment (jonatanschroeder) [inline exampleCourse/questions/demo/randomZippedFile/question.html:15]: Nit:
```suggestion
    Download the zipped file here:
```

**Review by jonatanschroeder — APPROVED** [review]

**Comment by coatless** [comment]
@eliotwrobson maybe change the name to:

```
demo/randomizedTemplateCodeFileZipped
```

from:

```
demo/randomZippedFile
```

to better convey demo contents.

**Pull request merged**